### ISSUE TITLE
Few options are not being initialised properly?

### ISSUE THREAD

**Issue by asheshv** [MohammadYounes/AlertifyJS#108]
Hi,

I am trying to override the options like:
onshow, onclose, onfocus, onmove, onmoved, onresize, onresized, onmaximize, onmaximized, onrestore, onrestored.

I can initialise them from 'main' function using the statement 'this.set('xxx', function() {...});', but - when passed as the options from the setup function, it does not get initialised.

I think the issue is with the file: 'https://github.com/MohammadYounes/AlertifyJS/blob/1.7.1/src/js/dialog/intro.js' function 'initialize', it does not have any initialization code after line# 296.

Please correct me (if I misunderstood the code).

Thanks,
Ashesh

**Comment by MohammadYounes** [comment]
Hi,

This is by design, [Setting Initialization](https://github.com/MohammadYounes/AlertifyJS/blob/1.7.1/src/js/dialog/intro.js#L271-L298) you referenced only sets values with a [Global Default](http://alertifyjs.com/guide.html#defaults).

Nevertheless, It makes sense to have the factory initialize them as these are shared by all dialogs. 

Thanks!

**Comment by asheshv** [comment]
It should be intialized as you mentioned,as these options are mentioned in the factory documentation.  http://alertifyjs.com/factory.html#factory.

**Comment by MohammadYounes** [comment]
That's a docs error, since they don't have a global default! I was thinking of initializing them based on `setup.options` only, such as:

```js
instance.set('onshow', setup.options.onshow);
```

Or should they be given a global default too?

```js
instance.set('onshow', setup.options.onshow === undefined ? alertify.defaults.events.onshow: setup.options.onshow);
```

**Comment by MohammadYounes** [comment]
Sorry for being late on this.

You are right. The factory should initialize all dialog options, The order would be:

1. User options (returned by `setup()` function)
2. Default options.

Will fix soon ... Thanks!

**Comment by MohammadYounes** [comment]
Fixed in 1.8

Thanks!
